capa/features/extractors/cape/thread.py

```diff
@@ -0,0 +1,67 @@
+# Copyright (C) 2020 Mandiant, Inc. All Rights Reserved.
+# Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at: [package root]/LICENSE.txt
+# Unless required by applicable law or agreed to in writing, software distributed under the License
+#  is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and limitations under the License.
+
+import logging
+from typing import Any, Dict, List, Tuple, Iterator
+
+from capa.features.common import Feature, String
+from capa.features.insn import API, Number
+from capa.features.address import Address
+from capa.features.extractors.base_extractor import ProcessHandle, ThreadHandle
+
+
+logger = logging.getLogger(__name__)
+
+
+def extract_call_features(behavior: Dict, ph:ProcessHandle, th: ThreadHandle) -> Iterator[Tuple[Feature, Address]]:
+    """
+    this method goes through the specified thread's call trace, and extracts all possible
+    features such as: API, Number (for arguments), String (for arguments).
+
+    args:
+      behavior: a dictionary of behavioral artifacts extracted by the sandbox
+      ph: process handle (for defining the extraction scope)
+      th: thread handle (for defining the extraction scope)
+
+    yields:
+      Feature, address; where Feature is either: API, Number, or String.
+    """
+
+    calls:List[Dict] = None
+    for process in behavior["processes"]:
+        if ph.pid == process["process_id"] and ph.inner["ppid"] == process["parent_id"]:
+            calls:List[Dict] = process
+
+    tid = str(th.tid)
+    for call in calls:
+        if call["thread_id"] != tid:
+            continue
+        yield Number(int(call["return"], 16)), int(call["caller"], 16)
```

Comment:
can you share an example of how a rule might use the number feature with a return value? just want to get that documented somewhere.

Reply:
maybe an anti-debug rule might use the following feature?

```yaml
- and:
  - api: IsDebuggerPresent
  - number: 0
```

Reply:
I think to do this we'd want a specific return value feature. Otherwise, I foresee confusion/conflicts around arguments/current rules (at least for the beginning).

Reply:
I agree, perhaps I should remove this for now.